### PR TITLE
Xeno healing rework and fixes

### PR DESCRIPTION
**Pull request by Rohesie**
Adjust damage verbs were used only by simple mobs and xenos, though they follow the logic of humans in where double max health damage kills someone. Xenos have a different condition to die: less than critical health, which is `-100`. This caused the edge case of the weaker of xenos, the Young Runner, having 100 life and being able to take at most 200 damage of each kind. In order to die it would require to be under -100, that is why no matter how many bullets it took, it would stay alive, unless if it took 1 damage of another kind as well, like fire.
Larvas had a snowflake critical health, bypassing this.

* Added procs to adjust and reset pain vars.
* Added resetting of halloss, pain, stagger and slowdown vars, plus maximum plasma, to rejuvenate()
* Xenos no longer share the healing procs with simple mobs, to adjust for their specificity.
* Xeno healing proc tweaked a bit. It's easier to die if in crit and off-weeds, but recovery while on crit in weeds is much improved.
* Larvas heal fast anywhere, as long as not in crit and out of weeds.
* Fixes xenos being unable to heal without a living queen.
* Shooting at xeno's head no longer does +14% extra damage. That was a bad idea we inherited.
* Clean a bit of code
* Action button icons get updated in the life proc regardless of plasma amount.
* Only the drone tree can, again, pick facehuggers.
* Xenos without a queen can heal again.
* Pain vars and procs are for carbon, not for all the living.
* update_action_button_icons() defined for those that have actions

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/damage_procs.dm [hunk 1]
```diff
@@ -108,13 +107,20 @@
 			adjustFireLoss(damage)
 
 	updatehealth()
-	return 1
+	return TRUE
+
+/mob/living/carbon/Xenomorph/adjustBruteLoss(var/amount)
```
Comment (ninjanomnom) [inline code/modules/mob/living/carbon/xenomorph/damage_procs.dm]: You don't need `var/` in args like this

File: code/modules/mob/living/carbon/xenomorph/damage_procs.dm [hunk 2]
```diff
@@ -108,13 +107,20 @@
...
+	bruteloss = CLAMP(bruteloss + amount, 0, maxHealth - crit_health)
+
+/mob/living/carbon/Xenomorph/adjustFireLoss(var/amount)
```
Comment (ninjanomnom) [inline code/modules/mob/living/carbon/xenomorph/damage_procs.dm]: ^

File: code/modules/mob/living/carbon/xenomorph/life.dm [hunk 3]
```diff
@@ -101,18 +95,40 @@ adjustFireLoss(-(maxHealth / 70 + 0.5 + (maxHealth / 70) * recovery_aura/2)*(m))
 		adjustFireLoss(fire_stacks + 3)
 
 /mob/living/carbon/Xenomorph/proc/handle_living_health_updates()
+	if(health >= maxHealth || hardcore) //no damage, don't bother
+		updatehealth()
```
Comment (ninjanomnom) [inline code/modules/mob/living/carbon/xenomorph/life.dm]: Why update health if nothing is changing?

File: code/modules/mob/living/carbon/xenomorph/update_icons.dm [hunk 4]
```diff
@@ -38,9 +35,13 @@
 			icon_state = "[caste] Running"
 		else
 			icon_state = "[caste] Walking"
-
 	update_fire() //the fire overlay depends on the xeno's stance, so we must update it.
 
+/mob/living/carbon/Xenomorph/proc/update_action_button_icons()
```
Comment (ninjanomnom) [inline code/modules/mob/living/carbon/xenomorph/update_icons.dm]: This seems like something worth having at the same place you can have actions instead of being a xenomorph specific proc

File: code/modules/mob/living/carbon/xenomorph/life.dm (hunk 3 above)
Comment (Rohesie) [reply code/modules/mob/living/carbon/xenomorph/life.dm]: `updatehealth()` does 3 things:
* Update health by recalculating it with the brute and fire loss.
* Update the medhud.
* Update the stat, killing the xeno if under a certain health, making it unconscious under another, and conscious over it.

It could be made more efficient, but the idea is that a xeno could transition from a state of low health to max, or one that warrants HUD changes. Suggestions?

File: code/modules/mob/living/carbon/xenomorph/update_icons.dm (hunk 4 above)
Comment (Rohesie) [reply code/modules/mob/living/carbon/xenomorph/update_icons.dm]: Made an `item`-specific one, that was the only other instance using it.

File: code/modules/mob/living/carbon/shock.dm [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /mob/living/var/traumatic_shock = 0
-/mob/living/carbon/var/shock_stage = 0
+/mob/living/var/shock_stage = 0
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/shock.dm]: why are these vars in a carbon file if they're for living?

Comment (Qustinnus) [reply code/modules/mob/living/carbon/shock.dm]: and why are they for living? the procs are for carbon.

File: code/modules/mob/living/carbon/xenomorph/damage_procs.dm [hunk 6]
```diff
@@ -93,13 +94,11 @@
 	if(damage > 12) //Light damage won't splash.
 		check_blood_splash(damage, damagetype, chancemod)
 
-	if(stat == DEAD) return
+	if(stat == DEAD)
+		return FALSE
 
-	if(warding_aura && damage > 0) //Damage reduction. Every half point of warding decreases damage by 2.5 %. Maximum is 25 % at 5 pheromone strength.
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/xenomorph/damage_procs.dm]: damage > 0 is useless

File: code/modules/mob/living/carbon/xenomorph/update_icons.dm (hunk 4 above)
Comment (ninjanomnom) [reply code/modules/mob/living/carbon/xenomorph/update_icons.dm]: It doesn't matter what else is currently using it, it should be defined at whatever level can have action buttons

File: code/modules/mob/living/carbon/xenomorph/damage_procs.dm (hunk 6 above)
Comment (Rohesie) [reply code/modules/mob/living/carbon/xenomorph/damage_procs.dm]: Yes, and it's removed.
